src/metadata.rs

```diff
@@ -62,7 +63,10 @@ pub fn fetch_metadata(provider: &str) -> errors::Result<Box<dyn providers::Metad
         "ibmcloud" => box_result!(IBMGen2Provider::try_new()?),
         // IBM Cloud - Classic infrastructure.
         "ibmcloud-classic" => box_result!(IBMClassicProvider::try_new()?),
-        "openstack-metadata" => box_result!(OpenstackProvider::try_new()?),
+        "openstack" => match OpenstackConfigDrive::try_new() {
```

Comment:
This should allow Afterburn to detect OpenStack and start scraping the metadata on first boot, though it's conflicting with the design decision in https://github.com/coreos/afterburn/issues/251.. Please verify @lucab

Reply:
I think we still need to support anyone who was previously explicitly injecting `openstack-metadata` as a provider, which includes the MCO since https://github.com/openshift/machine-config-operator/commit/76e932b3c369b5c77e970f56bd74b3361dd9352e

Reply:
Indeed, let's keep both `openstack` and `openstack-metadata` providers.